### PR TITLE
Implement some missing API handlers for the wallets endpoints

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#96

# Overview

- [x] I have extended wallet layer and API layer with 'listWallets' and 'deleteWallet' 
- [x] I have added proper tests to the DB layer to cover for the new 'deleteWallet' command

# Comments

### CODE REVIEW

File: test/unit/Cardano/Wallet/DB/MVarSpec.hs [hunk 1]
```diff
@@ -353,6 +386,22 @@ prop_createWalletTwice (key@(PrimaryKey wid), cp, meta) =
         runExceptT (createWallet db key cp meta) `shouldReturn` Right ()
         runExceptT (createWallet db key cp meta) `shouldReturn` Left err
 
+-- | Trying to create a same wallet twice should yield an error
```
Comment (piotr-iohk) [inline test/unit/Cardano/Wallet/DB/MVarSpec.hs:389]: I suppose should be _Trying to remove the same wallet twice..._

Comment (KtorZ) [reply test/unit/Cardano/Wallet/DB/MVarSpec.hs:389]: Indeed :+1: , wrong copy-pasting :grimacing:

Comment (piotr-iohk) [reply test/unit/Cardano/Wallet/DB/MVarSpec.hs:389]: BTW. Perhaps the case _create wallet twice_ could be added here also? Or maybe it is not handled on the DB layer (I mean the fact _the same_ wallet cannot be created multiple times)?

Comment (piotr-iohk) [reply test/unit/Cardano/Wallet/DB/MVarSpec.hs:389]: oh wait, it's already there :)

**Review by piotr-iohk — APPROVED** [review]
lgtm :+1:

**Comment by KtorZ** [comment]
@piotr-iohk Merging this, I've fixed the typo in a next incoming PR :+1:

**Pull request merged**